src/Tests/Microsoft.Diagnostics.Monitoring.Tool.FunctionalTests/ExceptionsTests.cs

```diff
@@ -0,0 +1,178 @@
+                appValidate: async (appRunner, apiClient) =>
+                {
+                    DateTime startTime = DateTime.UtcNow.ToLocalTime();
+
+                    await GetExceptions(apiClient, appRunner, ExceptionFormat.NewlineDelimitedJson);
+
+                    DateTime currentTime = DateTime.UtcNow.ToLocalTime();
+
+                    var exceptionsDict = JsonSerializer.Deserialize<Dictionary<string, object>>(exceptionsResult);
+
+                    Assert.Equal("2", exceptionsDict["id"].ToString());
+                    Assert.Equal(ExceptionType, exceptionsDict["typeName"].ToString());
+
+                    var timestamp = DateTime.Parse(exceptionsDict["timestamp"].ToString());
+                    Assert.True(startTime < timestamp);
+                    Assert.True(currentTime > timestamp);
+                    Assert.Equal(ModuleName, exceptionsDict["moduleName"].ToString());
+                    Assert.Equal(ExceptionMessage, exceptionsDict["message"].ToString());
+
+                    var callStackResultsRootElement = JsonSerializer.SerializeToDocument(exceptionsDict["callStack"]).RootElement;
+
+                    Assert.NotEqual("0", callStackResultsRootElement.GetProperty("threadId").ToString());
+
+                    var topFrame = callStackResultsRootElement.GetProperty("frames").EnumerateArray().FirstOrDefault();
+
+                    Assert.Equal(FrameMethodName, topFrame.GetProperty("methodName").ToString());
+                    Assert.Equal(2, topFrame.GetProperty("parameterTypes").GetArrayLength());
+                    Assert.Equal(FrameParameterType, topFrame.GetProperty("parameterTypes")[0].ToString());
+                    Assert.Equal(FrameParameterType, topFrame.GetProperty("parameterTypes")[1].ToString());
+                    Assert.Equal(FrameClassName, topFrame.GetProperty("className").ToString());
+                    Assert.Equal(FrameModuleName, topFrame.GetProperty("moduleName").ToString());
+                },
+                configureApp: runner =>
+                {
+                    runner.Architecture = targetArchitecture;
+                    runner.EnableMonitorStartupHook = enableStartupHook;
+                },
+                configureTool: runner =>
+                {
+                    runner.ConfigurationFromEnvironment.EnableInProcessFeatures();
+                });
+        }
+
+        private async Task GetExceptions(ApiClient apiClient, AppRunner appRunner, ExceptionFormat format)
+        {
+            await appRunner.SendCommandAsync(TestAppScenarios.Exceptions.Commands.Begin);
+
+            int processId = await appRunner.ProcessIdTask;
+
+            await RetryUtilities.RetryAsync(
+                () => CaptureExtensions(apiClient, processId, format),
+                shouldRetry: (Exception ex) => ex is ArgumentException,
+                maxRetryCount: 5,
+                outputHelper: _outputHelper);
+        }
```

Comment:
I don't see where `TestAppScenarios.Exceptions.Commands.End` is sent. I would assume it would be sent here.